lighthouse-cli/test/cli/run-test.js

```diff
@@ -21,14 +24,6 @@ const testConfig = {
   },
 };
 
-// Map plugin name to fixture since not actually installed in node_modules/.
-jest.mock('lighthouse-plugin-simple', () => {
```

Comment:
how does this work without this?

Reply:
somehow ``require.resolve('lighthouse-plugin-simple')` resolves to `/Users/cjamcl/src/lighthouse/lighthouse-core/test/fixtures/config-plugins/lighthouse-plugin-simple/plugin-simple.js` [here](https://github.com/GoogleChrome/lighthouse/blob/c53a030dab512c7dbafcd8a256b33ac541286a50/lighthouse-core/config/config-helpers.js#L444). Not sure how the module is registered... the `plugin-simple.js` bit at least is explained by that folder's `package.json` `main`.

Reply:
somehow `require.resolve('lighthouse-plugin-simple')` resolves to `/Users/cjamcl/src/lighthouse/lighthouse-core/test/fixtures/config-plugins/lighthouse-plugin-simple/plugin-simple.js` [here](https://github.com/GoogleChrome/lighthouse/blob/c53a030dab512c7dbafcd8a256b33ac541286a50/lighthouse-core/config/config-helpers.js#L444). Not sure how the module is registered... the `plugin-simple.js` bit at least is explained by that folder's `package.json` `main`.

Reply:
so weird, works here too. This completely goes against what I thought I knew about how resolve works :)

Reply:
just checked from master too, same there. this mock isn't needed for some reason unrelated to this PR. weeeeird

Reply:
this doesn't help but here:

```
{
        moduleIdentifier: 'lighthouse-plugin-simple',
        configDir: undefined,
        category: 'plugin',
        'require.resolve(moduleIdentifier)': '/Users/cjamcl/src/lighthouse/lighthouse-core/test/fixtures/config-plugins/lighthouse-plugin-simple/plugin-simple.js',
        'require.resolve.paths(moduleIdentifier)': [
          '/Users/cjamcl/src/lighthouse/lighthouse-core/config/node_modules',
          '/Users/cjamcl/src/lighthouse/lighthouse-core/node_modules',
          '/Users/cjamcl/src/lighthouse/node_modules',
          '/Users/cjamcl/src/node_modules',
          '/Users/cjamcl/node_modules',
          '/Users/node_modules',
          '/node_modules'
        ]
      }
```

Reply:
OK so it 100% fails when using the CLI directly, so this must be a jest side effect.

Reply:
didn't find much documentation on it, but looks like yeah, `jest` walks through your project and automatically finds all the `package.json` files. `jest-resolve` has a ModuleMap, which internally had a `_raw.map`, which looked like:

```
'gulp-lighthouse-recipe' => 'docs/recipes/gulp/package.json'
'custom-lighthouse-recipe' => 'docs/recipes/custom-audit/package.json'
'custom-lighthouse-pptr-recipe' => 'docs/recipes/custom-gatherer-puppeteer/package.json'
'lighthouse-plugin-example' => 'docs/recipes/lighthouse-plugin-example/package.json'
'lighthouse-logger' => 'lighthouse-logger/package.json'
'lighthouse' => 'package.json'
'lighthouse-plugin-config' => 'lighthouse-core/test/fixtures/config/lighthouse-plugin-config-helper/package.json'
'lighthouse-plugin-no-groups' => 'lighthouse-core/test/fixtures/config-plugins/lighthouse-plugin-no-groups/package.json'
'lighthouse-plugin-no-category' => 'lighthouse-core/test/fixtures/config-plugins/lighthouse-plugin-no-category/package.json'
'lighthouse-plugin-simple' => 'lighthouse-core/test/fixtures/config-plugins/lighthouse-plugin-simple/package.json'
'legacy-javascript' => 'lighthouse-core/scripts/legacy-javascript/package.json'
```

could be their haste modules, which apparently are ["`package.json` files outside of `node_modules` folders anywhere in the file system"](https://github.com/facebook/jest/blob/d7f097543f8c222b1ec52bc9bddf3a60c40d59ce/packages/jest-resolve/src/resolver.ts#L209-L210). I didn't verify that's the path triggering this but close enough :)

Maybe worth leaving a comment that jest automatically resolves the `'--plugins=lighthouse-plugin-simple'` flag to the correct location?